File: 351004/Bunevich/labs/task330/src/main/java/com/example/demo/repository/StoryRepository.java
package com.example.demo.repository;

import com.example.demo.models.Story;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.domain.Specification;

import java.util.Optional;

public interface StoryRepository extends BaseJpaRepository<Story> {
    @Override
    @EntityGraph(attributePaths = "tags")
    Optional<Story> findById(Long id);

    @Override
    @EntityGraph(attributePaths = "tags")
    Page<Story> findAll(Specification<Story> spec, Pageable pageable);

    boolean existsByWriterId(Long writerId);
}

File: 351003/Soika/entities-app/publisher/src/main/kotlin/com/example/entitiesapp/repository/StoryRepository.kt
package com.example.entitiesapp.repository

import com.example.entitiesapp.model.Story
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface StoryRepository : JpaRepository<Story, Long> {
    fun findAllByWriterId(writerId: Long): List<Story>
    fun existsByMarksId(markId: Long): Boolean
}